src/docs/arch-backup.dox

```diff
@@ -0,0 +1,60 @@
+/*! @arch_page arch-backup Backup
+
+# Overview #
+
+Backup in WiredTiger can be performed using the \c "backup:" cursors. WiredTiger
+supports four types of backups.
+
+1. Full database
+2. Block based incremental backup
+3. Log based incremental backup
+4. Target backup
+
+# Full database #
+
+When the application opens the backup cursor, internally WT collects all the tables
+that are present in the database and stores them in a list to let them be backed up.
+As part of the process of collecting the list of files to be backed up, WT takes the
+schema lock to block any modifications to create/drop new tables in the database.
+In addition to table creation, there is a contract that all files that exist at the
+time the backup starts exist for the duration of the backup.
+
+As part of backup, WT internally collects all the log files also that are present in the
+database and stores them in a list to let them to be backed up. These log files are
+required to make the backup consistent. As part of collecting the log files, forces a
+log file switch and it does not allow any log files to be archived/removed. WT doesn't
+use any pre allocated log files and create new set of log files.
```

Comment:
This paragraph feels a bit choppy. Maybe something like:
WiredTiger log files are also part of the generated file list for backup. There is a contract that the log files do not contain or make visible operations that are added to the log after the backup started that could have adverse affects during recovery and restore. Therefore WiredTiger forces a log file switch when opening the backup cursor. To fulfill the earlier contract that all files must exist during backup, WiredTiger does not use any pre-allocated log files nor does it archive or remove old log files during the time the backup cursor is open.

Reply:
Thanks, fixed as suggested.